src/auth/oauth.ts

```diff
@@ -146,6 +159,11 @@ export interface PushedAuthorizationRequest extends ClientCredentials {
    */
   code_challenge?: string;
 
+  /**
+   * Can carry fine-grained authorization data in OAuth messages as part of Rich Authorization Requests(RAR) {@link https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow/authorization-code-flow-with-rar | Reference}
+   */
+  authorization_details?: Required<AuthorizationDetails>[];
```

Comment:
Why do we need `Required<>` ? `AuthorizationDetails` does not have optional properties defined, so can't we just drop that?

Reply:
Type of `AuthorizationDetails` is :

```
interface AuthorizationDetails {
  type: string;
  [key: string]: any;
}
```

My understanding is `[key: string]: any;` property is considered as optional, we need not explicitly append it with `key: string]?: any`

Reply:
I dont think using Required has any use here, see https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgIIFcwAsD2VgBecYwOIAIhGHMADYDOyA3gFDLJgCeADhAFzJ6YfCADmAbjbIA2gGsInAUJGiAugLghOkgL4sWoSLEQoAKhCHIIAD0ggAJowBKEAI7pgUCPYA8GbHiExKQUVDQMAHzM+nosCGSW1gLmlgC80excvAIA5JBCOSyx8SCWimiYuPhEJGSU1HSM6ayZPPzIeRZghTpAA

Reply:
Yes Frederick, you are correct, my interpretation was erroneous. Thanks for taking time and providing an example.

After examining other requests, i concluded, this interface may not be required.
Since node-auth0 is a low level SDK, we will directly request the customer to supply AuthorizationDetails in required format.